x-pack/platform/plugins/shared/alerting/common/routes/schedule/validation/validation_schedule/v1.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+import { DURATION_REGEX, INTERVAL_FREQUENCY_REGEXP } from '../../constants';
+
+export const validateSchedule = (schedule: {
```

Comment:
Is the user allowed to define `duration` **and** `end date`?

Reply:
> Can the user define duration and end date?

yes

> How then do we validate the duration, does it have to match exactly the difference between the start and end date?

we compare duration only with every (interval-frequency). Should we compare it with end date? 🤔 @cnasikas

Reply:
> How then do we validate the duration, does it have to match exactly the difference between the start and end date?

`end` is optional, it gets converted to the `until` property in RRULE, meaning "recur until this date." Maybe we should rename `end` to `until` to reduce confusion? Making it part of `recurring.end` is a little helpful but I think if @adcoelho is getting confused then the user might also.

It's possible for the `duration` to go on longer than the `end` date, the rrule library won't throw an error or anything, it just won't make the rule recur ever.